tensorflow/core/kernels/unique_op.cc

```diff
@@ -237,6 +248,19 @@ class UniqueOp : public OpKernel {
       }
     }
   }
+
+ private:
+  void AutoAdjustUniqTableSize(const size_t uniq_size) {
+    if ((uniq_size << 1) > uniq_table_size_) {
+      uniq_table_size_ = uniq_size * 2.1;
+    } else if (uniq_size > (uniq_table_size_ << 2)) {
+      uniq_table_size_ = (uniq_size >> 2);
+    }
+  }
+
+ private:
+  size_t uniq_table_size_ = 0;
```

Comment:
Drive-by comment: Multiple threads could read and write this field during `Compute()` if there are concurrent `Run()` calls, so you'll at least need to guard it with a mutex (and I suspect that it's a hot enough path that you might want to put the effort into implementing a lock-free version using `std::atomic`).

Reply:
I copy code from tensorflow/core/kernels/dynamic_partition_op.cc
DynamicPartitionOp::num_partitions_ is a free-lock member variable .